app/code/Magento/CustomerGraphQl/Model/Customer/ValidateCustomerData.php

```diff
@@ -32,25 +41,40 @@ class ValidateCustomerData
      *
      * @param GetAllowedCustomerAttributes $getAllowedCustomerAttributes
      * @param EmailAddressValidator $emailAddressValidator
+     * @param CustomerMetadataInterface $customerMetadata
      */
     public function __construct(
         GetAllowedCustomerAttributes $getAllowedCustomerAttributes,
-        EmailAddressValidator $emailAddressValidator
+        EmailAddressValidator $emailAddressValidator,
+        CustomerMetadataInterface $customerMetadata
     ) {
         $this->getAllowedCustomerAttributes = $getAllowedCustomerAttributes;
         $this->emailAddressValidator = $emailAddressValidator;
+        $this->customerMetadata = $customerMetadata;
     }
 
     /**
      * Validate customer data
      *
      * @param array $customerData
-     *
-     * @return void
-     *
      * @throws GraphQlInputException
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
      */
     public function execute(array $customerData): void
+    {
+        $this->validateRequiredArguments($customerData);
```

Comment:
I like the idea here. Would be nice to have an interface that has a validate method to accept array $customerData and validator classes which implement the interface corresponding to arguments, email and gender. And you can pass in the validators via di.xml as arguments.

Reply:
@prabhuram93 
Could you please verify a new solution with defined validators in the `di.xml`?

Thank you!

Reply:
Cool. looks good!